pkg/grafana/datasources.go

```diff
@@ -164,6 +153,82 @@ func (s *Service) generateDatasources(organization *organization.Organization) (
 		}))
 	}
 
+	// 2. Create per-tenant datasources ONLY for alerting-enabled tenants
+	// Skip per-tenant datasources for mono-tenant organizations as they're redundant
+	alertingTenants := organization.GetAlertingTenants()
+	if len(organization.Tenants()) > 1 {
+		for _, tenant := range alertingTenants {
+			// Per-tenant Loki datasource for log viewing and alerting
```

Comment:
Since we create per tenant Mimir and Loki datasource for managing alerts I would recommend to explicitly set the `manageAlerts` field to `true` on those datasources, and set it to false on the main datasource when there are multiple tenants. This would avoid breaking the rules page for the multi tenant datasource. See https://github.com/giantswarm/observability-operator/pull/526/changes/12b65baa71a75762397acf86c75a49834a2f1eb0#diff-b9569c8b0317668d9c4d9dc897eede62bec5ff87a20004833fb64a80ed0348fcL170